blog/content/post/2020-02-05-meta-pcengines-xen.md

```diff
@@ -0,0 +1,140 @@
+abstract: 'Xen Project creates a software system that allows the execution of
+  multiple virtual guest operating systems simultaneously on a single physical machine.
+  In this case, it is a PC Engines apu2 platform.'
+cover: /covers/3mdeb-logo_simple-color.png
+author: norbert.kaminski
+layout: post
+published: false
+date: 2020-02-05
+archives: "2020"
+
+tags:
+  - XEN
+  - meta-pcengines
+  - apu2
+  - PC engines
+categories:
+  - OS Dev
+  - Security
+  - Manufacturing
+
+---
+### What is the Xen Project?
+
+Xen Project creates a software system, that allows the execution of multiple
+virtual guest environments simultaneously on a single physical machine.
+It is valuable when you want to create a complex security software system
+consisting of several operating systems. The unprivileged domains
+(domUs), such as virtual operating systems have no direct access to the
+hardware.
+
+Between software and hardware is a hypervisor. It allocates resources for
+new domains (the virtual operating systems) and schedules the existing ones.
+To learn more about hypervisor see the official
+[documentation](https://wiki.xenproject.org/wiki/Hypervisor).
+
+### What is Domain0?
+
+<i>
+    <h2>
+      <p align=center>Xen Project architecture </p>
+    </h2>
+</i>
+
+![Diagram](https://cloud.3mdeb.com/index.php/s/HNmc8yqbcCQzCje/preview)
+
+As shown in the diagram, Domain0 (Dom0) is the privileged domain and it manages
+unprivileged domains (domUs). Dom0 is started by the hypervisor on boot and
+ensures its usability. Therefore it is the Domain that has direct access to the
+hardware. The Domain0 provides hardware connections to DomUs. Dom0 allows full
+virtualized guests to use the virtual hardware and passes the physical links
+paravirtualized domains.
+
+### How to build Dom0 on the PC Engines apu2?
+
+To build the special domain for the PC Engines apu2 board, We will use the
+[meta-pcengines repository](https://github.com/3mdeb/meta-pcengines/tree/c4ee98ab390b073807173584107c09f49ac1e390).
+The repository contains the yocto layer, that allows you to create a
+minimal image of the dom0. To build a usable version of the dom0 image, the
+meta-pcengines needs a [kas tool](https://kas.readthedocs.io/en/0.19.0/index.html).
+Installation, configuration, and some more information about that tool
+you can find in the previous [blog](https://blog.3mdeb.com/2019/2019-02-07-kas/).
+
+Once you have installed the kas, clone the
+[meta-pcengines repository](https://github.com/3mdeb/meta-pcengines/tree/c4ee98ab390b073807173584107c09f49ac1e390).
+To reproduce my results, check if the source commit matches shown above.
+The source consists of configuration files, a kas script, and recipe files.
+The layer configuration specifies settings for the platform and distro.
+It points out where the recipes are placed. The kas script is used by the kas
+tool to clone and checkout `bitbake` layers. It also allows kas to create a
```

Comment:
OpenEmbedded layers or Yocto layers or OpenEmbedded/Yocto layers might be better.